benchmark/run.js

```diff
@@ -10,31 +10,54 @@ const cli = CLI(`usage: ./node run.js [options] [--] <category> ...
 
   --filter pattern          string to filter benchmark scripts
   --set    variable=value   set benchmark variable (can be repeated)
+  --format [simple|csv]     optional value that specifies the output format
 `, {
   arrayArgs: ['set']
 });
 const benchmarks = cli.benchmarks();
 
 if (benchmarks.length === 0) {
-  console.error('no benchmarks found');
-  process.exit(1);
+  console.error('No benchmarks found');
+  process.exitCode = 1;
+  return;
+}
+
+const validFormats = ['csv', 'simple'];
+const format = cli.optional.format || 'simple';
+if (!validFormats.includes(cli.optional.format)) {
+  console.error('Invalid format detected');
+  process.exitCode = 1;
+  return;
+}
+
+if (format === 'csv') {
+  console.log('"filename", "configuration", "rate", "time"');
 }
 
 (function recursive(i) {
   const filename = benchmarks[i];
   const child = fork(path.resolve(__dirname, filename), cli.optional.set);
 
-  console.log();
-  console.log(filename);
+  if (format !== 'csv') {
+    console.log();
+    console.log(filename);
+  }
 
   child.on('message', function(data) {
     // Construct configuration string, " A=a, B=b, ..."
     let conf = '';
     for (const key of Object.keys(data.conf)) {
       conf += ' ' + key + '=' + JSON.stringify(data.conf[key]);
     }
-
-    console.log(`${data.name}${conf}: ${data.rate}`);
+    // delete first space of the configuration
+    conf = conf.slice(1);
```

Comment:
This line can be removed if I reformat 57&59. I like this version better because I don't thing you should rely on spaces in strings you are concatenating.